management/server/peer.go

```diff
@@ -539,7 +540,7 @@ func (am *DefaultAccountManager) SyncPeer(ctx context.Context, sync PeerSync, ac
 
 	peer, updated := updatePeerMeta(peer, sync.Meta, account)
 	if updated {
-		err = am.Store.SaveAccount(ctx, account)
+		err = am.Store.SavePeer(ctx, account.Id, peer)
```

Comment:
If we just update the Meta why do not implement a SaveMeta instead of writing out the full Peer structure?

Reply:
We can't update only the meta since it is flattened by the tag `gorm:"embedded;embeddedPrefix:meta_"`, which creates a column per field. Plus, this method will be used more often.

Reply:
I think the gorm can handle the embedded structures well in case of an update just need pass it in the peer structure. Peer{Meta: peerSystemMeta}. But ok.